plans/autoAssign.md

```diff
@@ -0,0 +1,207 @@
+# Auto-Pair Feature for Open Water Reservations
+
+## Overview
+The Open Water Reservation System includes an Auto-Pair feature that intelligently matches reservation requests based on depth, equipment preferences, and buddy relationships. When a reservation request is submitted and approved by an admin, the system automatically pairs divers with compatible partners.
+
+## Core Algorithm (from /old_app/src/lib/autoAssign/)
+
+### Key Files:
+- **`createBuoyGroups.ts`** - Core pairing logic and depth compatibility
+- **`assignRsvsToBuoys.ts`** - Buoy assignment with equipment matching
+- **`assignHourlySpacesWithBreaks.ts`** - Space optimization algorithms
+- **`hourlyUtils.ts`** - Helper functions for grouping and assignments
+
+### Pairing Strategy:
+
+#### 1. **Buddy Group Formation**
+- Groups divers who are already buddies (pre-existing relationships)
```

Comment:
this statement is wrong. there is no such thing as friends/connection in the app. this will make the app assume the wrong db design